WordPress/WordPressTest/ContextManagerMock+AutomaticTeardown.swift

```diff
@@ -0,0 +1,72 @@
+import XCTest
+
+@testable import WordPress
+
+extension ContextManagerMock {
+    /// Override `ContextManager.shared` with the mock instance, and un-override it when
+    /// given `testCase` finishes.
+    ///
+    /// This method should be called during test run, typically from `XCTestCase` subclass's
+    /// `setUp` method or test methods.
```

Comment:
With the latest round of changes, I'm not sure it's appropriate to recommend calling this from a test method. After all, the cleanup is done only during the test _case_ `tearDown`. That is, once the test method finishes the `ContextManager.shared` will still be the `ContextManagerMock` instance.

The method name and input parameter do a clear job at explaining this, but this bit of documentation might be confusing.

(Of course, the overall problem is that we need to set global state in the first place, but that's a bigger and out of scope things to tackle).

```suggestion
    /// This method should be called during test set up, typically from `XCTestCase` subclass's
    /// `setUp` method.
```